src/me/mrCookieSlime/Slimefun/api/Soul.java

```diff
@@ -21,6 +21,12 @@ public static void storeItem(UUID uuid, ItemStack drop) {
 	public static void retrieveItems(Player p) {
 		if (Variables.soulbound.containsKey(p.getUniqueId())) {
 			for (ItemStack item: Variables.soulbound.get(p.getUniqueId())) {
+				if (item.equals(p.getInventory().getHelmet())) return;
+				if (item.equals(p.getInventory().getChestplate())) return;
+				if (item.equals(p.getInventory().getLeggings())) return;
+				if (item.equals(p.getInventory().getBoots())) return;
+				if (item.equals(p.getInventory().getItemInOffHand())) return;
```

Comment:
return; terminates the entire method which basically cancels all items if just one is equal. Use continue; instead